### PR TITLE
reduce the number of STM transactions

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Agent.hs [hunk 1]
```diff
@@ -2133,7 +2133,7 @@ cleanupManager c@AgentClient {subQ} = do
       step <- asks $ cleanupStepInterval . config
       liftIO $ threadDelay step
     -- we are catching it to avoid CRITICAL errors in tests when this is the only remaining handle to active
-    waitActive a = liftIO (E.tryAny . atomically $ waitUntilActive c) >>= either (\_ -> pure ()) (\_ -> void a)
+    waitActive a = liftIO (E.tryAny $ waitUntilActive c) >>= either (\_ -> pure ()) (\_ -> void a)
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:2136]: this is suspicious - why does it need "try"?

Comment (spaced4ndy) [reply src/Simplex/Messaging/Agent.hs:2136]: maybe some older wrong refactoring? not sure. we'd only want to try separate steps and proceed onto next

**Review by spaced4ndy — APPROVED** [review]

File: src/Simplex/Messaging/Agent.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Agent.hs:2136]: no, waitUntilActive can't throw

Comment (epoberezkin) [reply src/Simplex/Messaging/Agent.hs:2136]: ah - there is some comment...

**Pull request merged**